tests/system/test_sqlalchemy_bigquery.py

```diff
@@ -774,3 +774,27 @@ def test_unnest(engine, bigquery_dataset):
         f" unnest(`{bigquery_dataset}.test_unnest_1`.`objects`) AS `foo_objects`"
     )
     assert sorted(r[0] for r in conn.execute(query)) == ["a", "b", "c", "x", "y"]
+
+
+@pytest.mark.skipif(
+    packaging.version.parse(sqlalchemy.__version__) < packaging.version.parse("1.4"),
+    reason="regexp_match support requires version 1.4",
+)
+def test_regexp_match(session, table):
+    results = session.query(table.c.string).where(
+        table.c.string.regexp_match(".*foo.*")
```

Comment:
hey folks 👋 I might need some help here for more specific regex pattern matching

Reply:
@TeddyCr If I am tracking this correctly based on the following pytest fixture, it appears that the table we match up against gets generated in the `table` method and uses the data in [sample.json](https://github.com/googleapis/python-bigquery-sqlalchemy/blob/f05979353f925b92def135f556fd267f1b71a190/tests/system/data/sample.json) 

```
@pytest.fixture(scope="session")
def table(engine, bigquery_dataset):
    return Table(f"{bigquery_dataset}.sample", MetaData(bind=engine), autoload=True)
```

It appears we are making the dataset here... 
https://github.com/googleapis/python-bigquery-sqlalchemy/blob/f05979353f925b92def135f556fd267f1b71a190/tests/system/conftest.py#L73

using the `load_sample_data()` method:
https://github.com/googleapis/python-bigquery-sqlalchemy/blob/f05979353f925b92def135f556fd267f1b71a190/tests/system/conftest.py#L38

Long story short... in looking at the data in the string column found in the sample.json file, it appears that we have a several options for expressions to search for. For example, we could search for `Bleecker`:

`table.c.string.regexp_match(".*Bleecker.*")`